### PR TITLE
[Feature]: support image visualization for tensorboard and wandb

### PR DESCRIPTION
**Pull request by Cathy0908**
Tensorboard visulization is as follows:

image

Wandb visulization is as follows:

image

The step interval shown in the figure is the number of steps of N epoch, N is the  eval intervel.

Usage:

```python
# evaluation
...
eval_config = dict(
    interval=1,
    visualization_config=dict(
        vis_num=10,  # number of images to show
        score_thr=0.5,  # boxes scores below the score_thr will be filtered out 
    ) 
)
...
log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook'),
        dict(type='TensorboardLoggerHookV2'),
        dict(type='WandbLoggerHookV2'),
    ])
```

### CODE REVIEW

File: benchmarks/extract_dataset_configs/imagenet.py [hunk 1]
```diff
@@ -23,4 +23,5 @@
             dict(type='CenterCrop', size=224),
             dict(type='ToTensor'),
             dict(type='Normalize', **img_norm_cfg),
+            dict(type='Collect', keys=['img', 'gt_labels'])
```
Comment (Cathy0908) [inline benchmarks/extract_dataset_configs/imagenet.py:26]: has nothing to do with this pr, fix previous bug, please refer to: https://github.com/alibaba/EasyCV/pull/6

File: easycv/core/visualization/image.py [hunk 2]
```diff
@@ -1,18 +1,119 @@
...
+    for label in labels:
+        # roughly estimate the proper font size
+        text_size, text_baseline = cv2.getTextSize(label,
+                                                   cv2.FONT_HERSHEY_DUPLEX,
+                                                   font_scale, thickness)
+
+        org = (text_baseline + text_size[1],
+               text_baseline + text_size[1] + cur_height)
+
+        # support chinese text
```
Comment (Cathy0908) [inline easycv/core/visualization/image.py:88]: support chinese font

File: easycv/hooks/tensorboard.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                if isinstance(img, np.ndarray):
+                    img = torch.from_numpy(img)
+                else:
+                    assert isinstance(
+                        img, torch.Tensor
+                    ), 'Only support np.ndarray and torch.Tensor type!'
+
+                filename = img_metas[i]['ori_filename']
+                self.writer.add_image(
+                    f'{vis_key}/{filename}',
```
Comment (Cathy0908) [inline easycv/hooks/tensorboard.py:31]: add group_name `vis_key`

File: easycv/core/visualization/image.py [hunk 4]
```diff
@@ -1,18 +1,119 @@
...
 import cv2
 import mmcv
 import numpy as np
 from mmcv.utils.misc import deprecated_api_warning
+from PIL import Image, ImageDraw, ImageFont
+from torch.hub import DEFAULT_CACHE_DIR
+
+
+def download_font(save_path=None):
+    url_path = 'http://pai-vision-data-hz.oss-accelerate.aliyuncs.com/EasyCV/pkgs/simhei.ttf'
```
Comment (wenmengzhou) [inline easycv/core/visualization/image.py:16]: add this font to git and auto package it into our python whl

File: easycv/hooks/tensorboard.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+            for i, img in enumerate(images):
+                if isinstance(img, np.ndarray):
+                    img = torch.from_numpy(img)
+                else:
+                    assert isinstance(
+                        img, torch.Tensor
+                    ), 'Only support np.ndarray and torch.Tensor type!'
+
+                filename = img_metas[i]['ori_filename']
```
Comment (wenmengzhou) [inline easycv/hooks/tensorboard.py:29]: is filename sufficient for visualization?   img_metas is not straightforward

File: configs/detection/yolox/yolox_s_8xb16_300e_coco.py [hunk 6]
```diff
@@ -168,7 +175,8 @@
     interval=100,
     hooks=[
         dict(type='TextLoggerHook'),
-        dict(type='TensorboardLoggerHook')
+        dict(type='TensorboardLoggerHookV2'),
+        # dict(type='WandbLoggerHookV2'),
```
Comment (wenmengzhou) [inline configs/detection/yolox/yolox_s_8xb16_300e_coco.py:179]: wandb config example should be added

File: easycv/core/visualization/image.py (hunk 4 above)
Comment (Cathy0908) [reply easycv/core/visualization/image.py]: put simhei.ttf to resource dir

File: easycv/hooks/tensorboard.py (hunk 5 above)
Comment (Cathy0908) [reply easycv/hooks/tensorboard.py]: add doc, img_metas is optional

File: configs/detection/yolox/yolox_s_8xb16_300e_coco.py (hunk 6 above)
Comment (Cathy0908) [reply configs/detection/yolox/yolox_s_8xb16_300e_coco.py:179]: add comment

File: easycv/hooks/tensorboard.py (hunk 5 above)
Comment (wenmengzhou) [reply easycv/hooks/tensorboard.py]: if imag_metas does not exsits in vis_result, line 29 will throw exception

File: configs/detection/yolox/yolox_s_8xb16_300e_coco.py (hunk 6 above)
Comment (wenmengzhou) [reply configs/detection/yolox/yolox_s_8xb16_300e_coco.py:179]: wandb initialization config is missing

File: easycv/hooks/wandb.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                ), 'Output `images` and `img_metas` must keep the same length!'
+
+            examples = []
+            for i, img in enumerate(images):
+                assert isinstance(img, np.ndarray)
+                img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+                pil_image = PILImage.fromarray(img, mode='RGB')
+                filename = img_metas[i].get(
+                    'ori_filename',
+                    'image') if img_metas is not None else 'image'
```
[the remark this answers is not in the document]
Comment (Cathy0908) [reply easycv/hooks/wandb.py]: done

**Review by wenmengzhou — APPROVED** [review]

**Pull request merged**